### DISCUSSION TITLE
Compilation with libtorch

### DISCUSSION THREAD

**Discussion by aiskhak** [idaholab/moose#22631] [Q&A Modules: General] [answered]
I would like to use libtorch capabilities within MOOSE.

I am using linux machine and followed the installation instructions for HPC:
https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html

Then I installed libtorch:
https://mooseframework.inl.gov/getting_started/installation/install_libtorch.html

I am able to execute tests:
./run_tests -j 8 --re libtorch_nn

However, when I create a custom app and enable in the Makefile the Stochastic Tools module:
STOCHASTIC_TOOLS := yes
it kind of does not see the module and as a result, stochastic tools do not work.

SO in general the problem is that the custom app does not see individal modules, while 

ALL_MODULES := yes

works. How to fix this?

**Comment by GiudGiud** [comment]
@grmnptr

**Comment by grmnptr** [comment, marked as answer]
Could you write here down step by step what you are doing? I mean during the installation of the custom app.

**Reply by aiskhak** [reply]
sorry, the problem was is that I did not understand the meaning of:

ifeq ($(ENABLE_LIBTORCH),true)
  app_non_unity_dirs = %src/surrogates
endif
